docs/specs/fee-calculator.md

```diff
@@ -0,0 +1,379 @@
+      name: string;
+      daily_totals: Array<{
+        date: string;
+        start_balance_wei: string;
+        end_balance_wei: string;
+        balance_change_wei: string;
+        distributions_wei: string;
+        distributions_count: number;
+        total_wei: string;
+      }>;
+    };
+  };
+}
+```
+
+## 8. Implementation Requirements
+
+### Validation Rules
+
+1. **Address Validation**:
+
+   - All addresses must be valid Ethereum addresses
+   - Addresses normalized to checksummed format
+
+2. **Date Validation**:
+
+   - All dates in ISO format (YYYY-MM-DD)
+   - Dates must be chronologically ordered
+
+3. **Amount Validation**:
+
+   - All amounts must be valid numeric strings
+   - Handle large numbers using bigint internally
+   - Convert back to strings for output
+
+4. **Data Completeness**:
+   - Throw error if critical data is missing
+
+### Security Considerations
+
+1. **FileManager ensures atomic file operations**
+
+2. **Error Handling**:
+   - Never expose system paths in error messages
+   - Fail gracefully with descriptive error messages
+
+## 9. Test Data Requirements
+
+### Minimum Test Data
+
+Integration tests require real blockchain data from Arbitrum Nova:
+
+1. **Distributors**: Use actual reward distributor addresses from test data:
+
+   - At least 5 distributors with varying creation dates
+   - Mix of high-volume and low-volume distributors
+   - Include distributors created at different times
+
+2. **Balance Data**: For each test distributor:
+
+   - Complete balance history for test period (minimum 30 days)
+   - Include days with increasing and decreasing balances
+   - Cover edge cases: zero balances, large balances
+
+3. **Distribution Events**: For each test distributor:
+
+   - All distribution events during test period
+   - Include days with multiple distributions
+   - Include days with no distributions
+   - Varying distribution amounts
+
+4. **Test Scenarios**:
+   - Distributor with consistent daily totals
+   - Distributor with sporadic activity
+   - Distributor with no distributions (totals from balance changes only)
+   - Distributor created mid-period
+   - Full pipeline test: all distributors for complete date range
+
+### Available Test Data
+
+Test data available in `__tests__/test-data/`:
+
+- **Distributor list**: Available in `distributor-detector/distributor-creation-events-raw.json` with 5 distributors
+- **Historical balance snapshots**: Available in `distributor-detector/balance_data/` for all 5 distributors across 9 dates (2022-07-11 to 2023-03-17)
+- **Distribution event records**: Available in `recipient-recieved/` directory with RecipientRecieved events for 3 of the 5 distributors (320 total events)
+
+### Missing Test Data
+
+The following test data needs to be generated:
+
+- **Expected total calculation results**: JSON files with pre-calculated expected daily totals for each distributor to validate the fee calculator output. This should include:
```

Comment:
Lets add that we'll write an independent script to try to generate this from the test data. When tickets are created from the spec this should be one of the first tickets